### PR TITLE
FISH-6556 Concurrency Fix Web Profile

### PR DESCRIPTION
**Pull request by aubi**
## Description
Payara failed the Concurrency TCK 3.0.2. It is fixed now:
* properly handle context lists during context switch (fix in ContextSetupProviderImpl.java)
* cancel scheduled Future (upgrade of patched Concurrency RI)

### Blockers
This patch depends on new version of patched Concurrency RI: https://github.com/payara/patched-src-concurrency-ri/pull/2 
The version 3.0.0.payara-p2 is uploaded to Payara's nexus.

### Testing Performed
Run latest Concurrency TCK, version 3.0.2. Updated runner: https://github.com/payara/jakartaee-10-tck-runners/pull/44

### Testing Environment
OpenJDK 11, Linux

### CODE REVIEW

File: appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java [hunk 1]
```diff
@@ -388,7 +388,7 @@ public void reset(ContextHandle contextHandle) {
         if (handle.getSecurityContext() != null) {
             SecurityContext.setCurrent(handle.getSecurityContext());
         }
-        if (handle.getInvocation() != null) {
+        if (handle.getInvocation() != null /*&& !handle.isUseTransactionOfExecutionThread()*/) {
```
Comment (breakponchito) [inline appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java:391]: Why adding this comment on the condition?

Comment (aubi) [reply appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java]: removed

**Comment by aubi** [comment]
Jenkins test please

**Review by breakponchito — APPROVED** [review]
LGTM

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by aubi** [comment]
Jenkins test please

**Pull request merged**